### PR TITLE
OSCI: Onboard non groovy e2e tests

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

This PR adds support for 'non-groovy' e2e tests to run in OpenShift CI. The companion OSCI PR is https://github.com/openshift/release/pull/29237.

### Changed roxctl proxy test to use localhost rather than central-proxy.stackrox.local

In OSCI the test runner has limited rights and cannot write to `/etc/hosts`. Therefore the existing approach of resolving `central-proxy.stackrox.local` to `127.0.0.1` with a `/etc/hosts` entry is not an option. Using HOSTALIASES, which worked as an option for ui-tests, does not appear to be supported by roxctl. So this PR uses `localhost` as the CN for tests/scripts/setup-certs.sh.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

CI is sufficient:
- [x] rehearsal jobs in OpenShift CI: https://prow.ci.openshift.org/pr-history/?org=openshift&repo=release&pr=29237
- [x] existing Circle CI: https://app.circleci.com/pipelines/github/stackrox/stackrox/14112/workflows/c9efa6e9-0374-4381-bb6a-cfbfb5bb4100
- [x] existing OpenShift CI: https://prow.ci.openshift.org/pr-history/?org=stackrox&repo=stackrox&pr=2013

After configurable change:
- [x] rehearsal jobs in OpenShift CI: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/29237/rehearse-29237-pull-ci-stackrox-stackrox-gavin-OSCI-onboard-non-groovy-gke-nongroovy-e2e-tests/1536948903214059520

### CODE REVIEW

**Review by sbostick — APPROVED** [review]
LGTM

File: scripts/ci/proxy/deploy.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ kubectl apply -f "${dir}/nginx-proxy-plain-http.yaml"
 cert_dir="${PROXY_CERTS_DIR:-$(mktemp -d)}"
 
 echo "Cert directory: ${cert_dir}"
-"${dir}/../../../tests/scripts/setup-certs.sh" "${cert_dir}" "central-proxy.stackrox.local" "Proxy CA"
```
Comment (SimonBaeumer) [inline scripts/ci/proxy/deploy.sh:21]: The test coverage is not affected by using `localhost` as long as Sensor does not connect to central via the proxy.
This is not the case in our e2e-tests. Our proxy tests are tested only inside the cluster and don't rely on DNS names set from the outside.

File: scripts/ci/proxy/nginx-proxy-tls-grpc.conf [hunk 2]
```diff
@@ -10,7 +10,7 @@ server {
 
 server {
 	listen 8443 ssl http2;
-	server_name central-proxy.stackrox.local;
```
Comment (SimonBaeumer) [inline scripts/ci/proxy/nginx-proxy-tls-grpc.conf:13]: Can you make this configurable that the parameter inserted from the outside is used as a server name here?
These scripts are very useful to quickly setup proxies for local testing.

File: scripts/ci/proxy/nginx-proxy-tls-http1.conf [hunk 3]
```diff
@@ -10,7 +10,7 @@ server {
 
 server {
 	listen 8443 ssl;
-	server_name central-proxy.stackrox.local;
```
Comment (SimonBaeumer) [inline scripts/ci/proxy/nginx-proxy-tls-http1.conf:13]: parameterize?

File: scripts/ci/proxy/nginx-proxy-tls-http2.conf [hunk 4]
```diff
@@ -10,7 +10,7 @@ server {
 
 server {
 	listen 8443 ssl http2;
-	server_name central-proxy.stackrox.local;
```
Comment (SimonBaeumer) [inline scripts/ci/proxy/nginx-proxy-tls-http2.conf:13]: parameterize?

File: scripts/ci/proxy/nginx-proxy-tls-multiplexed.conf [hunk 5]
```diff
@@ -10,7 +10,7 @@ server {
 
 server {
 	listen 8443 ssl http2;
-	server_name central-proxy.stackrox.local;
```
Comment (SimonBaeumer) [inline scripts/ci/proxy/nginx-proxy-tls-multiplexed.conf:13]: parameterize?

**Pull request merged**